src/Bicep.Core/Syntax/AccessExpressionSyntax.cs

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Parsing;
+
+namespace Bicep.Core.Syntax;
+
+/// <summary>
+/// Represents a syntax node that accesses a property or index of a base value.
+/// </summary>
+public abstract class AccessExpressionSyntax : ExpressionSyntax
+{
+    protected AccessExpressionSyntax(SyntaxBase baseExpression, Token? safeAccessMarker)
+    {
+        AssertTokenType(safeAccessMarker, nameof(safeAccessMarker), TokenType.Question);
+
+        BaseExpression = baseExpression;
+        SafeAccessMarker = safeAccessMarker;
+    }
+
+    public SyntaxBase BaseExpression { get; }
+
+    public Token? SafeAccessMarker { get; }
+
+    public abstract SyntaxBase AccessExpression { get; }
```

Comment:
Is this used for anything? I couldn't spot any places.

Reply:
This was vestigial from an earlier iteration. Will prune